enhancements/security/x509-trust.md

```diff
@@ -0,0 +1,213 @@
+  - "/enhancements/kube-apiserver/tls-config.md"
+  - "/enhancements/proxy/global-cluster-egress-proxy.md"
+---
+
+# X.509 Trust
+
+This enhancement provides a big-picture overview of X.509 trust management for cluster components.
+This enhancement provides for both cluster-scoped defaults and per-component overrides.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Several cluster components may make TLS requests to external services.
+For example, [the cluster-version operator][cluster-version-operator] may request available updates, [the machine-API operator][machine-api-operator] may call platform APIs to manage machines, [CRI-O][cri-o] may contact container image registries when downloading images for pods, etc.
+Cluster components may also perform other X.509 verification besides TLS requests.
+For example, user mail processing components may perform [S/MIME][smime] verification.
+This enhancement provides a framework for managing X.509 trust for these situations.
+
+Currently the default trust for CRI-O, the kubelet's cloud provider, and other node-namespace egress is the node's X.509 certificate authority (CA) store, which depends on the operating system ([Red Hat Enterprise Linux CoreOS (RHCOS)][rhcos], etc.) and any subsequent configuration ([the machine-config operator][machine-config-operator], etc.).
+The default trust for containers like the machine-API operator is that container's CA store, which depends on the base layer ([Red Hat Universal Base Image (UBI)][ubi], etc.).
+Many external components will have X.509 certificates signed by one of those default CAs, but this is not always the case.
+
+There may also be internal communication between cluster components that relies on X.509 trust but lacks a custom 
+The framework from this enhancement can also be used to configure trust for that communication as well.
+
+This enhancement provides a mechanism for configuring additional or alternative trust bundles for all cluster services that do not already have existing trust configuration mechanisms.
+
+## Motivation
+
+### Goals
+
+Administrators will have a single location to configure a default additional trust bundle to be used by all cluster components.
+They will also be able to specify whether this default is to be use in addition to, or instead of, the component's default trust bundle.
+
+Administrators will have per-component configuration locations to adjust the cluster-scoped default.
+As for the cluster-scoped default, they will be able to specify whether the per-component trust bundle is to be used in addition to, or instead of, the cluster-scoped default.
+
+### Non-Goals
+
+This enhancement does not address components making intra-cluster TLS requests.
```

Comment:
If a component makes a request to a route, is that considered intra-cluster? For example, auth-operator tries connecting to the auth route.

Reply:
really interesting question... I think of Ingress/Route as supporting cluster-external traffic by definition

Reply:
I stepped into the intra-cluster space a bit with [this pivot][1].  Moved away from the intra-cluster wording here with 4a3930d -> d6cd880.  But there's an existing mechanism to distribute trust in the route-signing CA, isn't there?  If so, folks hitting routes would stick with that existing mechanism and not use this new one.

[1]: https://github.com/openshift/enhancements/pull/115#discussion_r348210989